app/models/shipping_rate.rb

```diff
@@ -0,0 +1,61 @@
+require 'active_shipping'
+
+class ShippingRate
+  include ActiveModel::Validations
+
+  attr_accessor :origin, :destination, :package
+
+  validates :origin, :destination, :package, presence: true
+  validate :valid_locations
+  validate :valid_package
+
+  def initialize(attributes = {})
+    attributes.each do |name, value|
+      send("#{name}=", value)
+    end
+  end
+
+  def ups_rates # TODO: put in actual credentials
+    ups = ActiveShipping::UPS.new(login: ENV["ACTIVESHIPPING_UPS_LOGIN"], password: ENV["ACTIVESHIPPING_UPS_PASSWORD"], key: ENV["ACTIVESHIPPING_UPS_KEY"])
+    format_ups_rates(get_rates_from_shipper(ups))
+  end
+
+  def usps_rates
+    usps = ActiveShipping::USPS.new(login: ENV['ACTIVESHIPPING_USPS_LOGIN'])
+    format_usps_rates(get_rates_from_shipper(usps))
+  end
+
+  private
+
+  def format_ups_rates(rates)
+    rates.sort_by(&:price).map do |rate|
```

Comment:
If you are using `total_price` in the returned value, should this sort by that field instead?